docs/support/tutorials/nextflow-puhti.md

```diff
@@ -1,191 +1,309 @@
 
-### Bring your WtP singularity images to Puhti
+### 6.2 Bring your WtP Apptainer images to Puhti
 
-WtP is a multi-container pipeline requiring as many as 21 singularity images 
-(see [here for further details)](https://github.com/replikation/What_the_Phage/blob/master/configs/container.config) 
-at the time of writing this tutorial. All these containers were downloaded 
-in cPouta environment to avoid any build failures of singularity images on Puhti due to the lack 
-of privileged access for users. For the sake of this tutorial, the downloaded singularity 
-images were then uploaded to ``Allas`` which is an object storage environment at CSC.
+WtP is a multi-container pipeline requiring as many as 21 container images
+([more details here](https://github.com/replikation/What_the_Phage/blob/master/configs/container.config))
+at the time of writing this tutorial. All these containers were downloaded
+to a cPouta environment to avoid any build failures of container images on
+Puhti due to the lack of root privileges for users. For the sake of this
+tutorial, the downloaded container images were then uploaded to
+[Allas](../../data/Allas/introduction.md) which is an object storage
+environment at CSC.
+
+The images in Allas object storage can be downloaded to your project scratch as
+below:
 
-The images on `Allas` object storage can be downloaded to your project directory on scratch as below:
 ```bash
-mkdir -p /scratch/project_xxx/What_the_Phage/singularity
-cd /scratch/project_xxx/What_the_Phage/singularity
+mkdir -p /scratch/<project>/What_the_Phage/singularity
+cd /scratch/<project>/What_the_Phage/singularity
 wget https://a3s.fi/puhti_singularity/WtP_singularity.tar.gz
 tar -xavf WtP_singularity.tar.gz
 ```
 
-### Run WtP pipelines as a batch job on Puhti:
+### 6.3 Run WtP pipelines as a batch job on Puhti
 
-Submit the following batch script to run nextflow pipeline:
-```
+Submit the following batch script to run the Nextflow pipeline:
+
+```bash
 #!/bin/bash
 #SBATCH --time=01:00:00
 #SBATCH --partition=small
-#SBATCH --account=project_XXX
+#SBATCH --account=<project>
 #SBATCH --cpus-per-task=4
 
-export TMPDIR=$PWD
-
-# Activate  Nextflow on Puhti
+export TMPDIR=${PWD}
+export SCRATCH=/scratch/<project>/What_the_Phage
 
+# Activate Nextflow on Puhti
 module load nextflow/21.10.6 
 
 # Nextflow command here
-nextflow run /scratch/project_xxx/What_the_Phage/phage.nf --fasta /scratch/project_xxx/What_the_Phage/test-data/OX2_draft.fa --cores 4 --output results -profile local,singularity --cachedir /scratch/project_xxx/What_the_Phage/singularity --databases /scratch/project_xxx/What_the_Phage/databases/WtP_databases --workdir /scratch/project_xxx/What_the_Phage/workflow-phages-username 
+nextflow run ${SCRATCH}/phage.nf \
+         --fasta ${SCRATCH}/test-data/OX2_draft.fa \
+         --cores 4 \
+         --output results \
+         --profile local,singularity \
+         --cachedir ${SCRATCH}/singularity \
```

Comment:
Should it be singularity or apptainer?

Reply:
for this pipeline and version, it is "singularity"